apis/v1alpha2/httproute_types.go

```diff
@@ -853,5 +853,5 @@ type HTTPBackendRef struct {
 
 // HTTPRouteStatus defines the observed state of HTTPRoute.
 type HTTPRouteStatus struct {
-	RouteStatus `json:",inline"`
+	RouteStatusWithHostnames `json:",inline"`
```

Comment:
Why not keep the name of the struct the same? will we also have RouteStatusWithoutHostnames?

Reply:
Good question - this is because TCPRoute and UDPRoute don't support hostnames so we're leaving that out of status. So RouteStatusWithHostnames is really just that, it is a type that inlines RouteStatus + adds a hostnames field.

Reply:
Does it make sense to pull up the Hostnames field instead of creating a new type? It is all inline in any event.

Reply:
So just add the hostnames field to both HTTPRouteStatus and TLSRouteStatus? Although that might be simpler now, I think we'd regret it as more routes like GRPCRoute are added.

Reply:
Not clear why writing it out is better/worse to be honest -- this makes the usage of the API for the go consumer somewhat cumbersome (there are multiple levels of inline being collapsed into the YAML representation).  

I don't think there is a requirement to keep the status struct the same Go types between the route types?

Reply:
I dug into this further, and it's actually substantially more complicated to split this out than I'd thought.

Essentially RouteStatus includes a list of RouteParentStatus structs, and we want to add a Hostnames field to the RouteParentStatus structs. The RouteParentStatus struct is sufficiently complex that I don't think we should duplicate it. So we could potentially replace RouteStatusWithHostnames with something like this for each route type that includes hostnames (currently HTTP and TLS, will likely include GRPC soon):

```go
HTTPRouteStatus struct {
	// Parents is a list of parent resources (usually Gateways) that are
	// associated with the route, and the status of the route with respect to
	// each parent. When this route attaches to a parent, the controller that
	// manages the parent must add an entry to this list when the controller
	// first sees the route and should update the entry as appropriate when the
	// route or gateway is modified.
	//
	// Note that parent references that cannot be resolved by an implementation
	// of this API will not be added to this list. Implementations of this API
	// can only populate Route status for the Gateways/parent resources they are
	// responsible for.
	//
	// A maximum of 32 parent resources will be represented in this list. An
	// empty list means the route has not been attached to any parent resource.
	//
	// +kubebuilder:validation:MaxItems=32
	Parents []RouteParentStatusWithHostnames `json:"parents"`
}
```

That would make for a larger diff and would increase the chance that one of these got out of sync. I'm not sure it's worth the change, but I may be missing a better way to do it.

Reply:
In [route delegation status draft](https://hackmd.io/@youngnick/gateway-api-statusl) I asked:

> If we go with RouteParentStatusWithHostnames, would we add this per-rule attachment status list there so that it only applies to HTTP and TLS?
>
> Or if we went with HTTPRouteParentStatus, would we add it there so it applies only to HTTP?
>
> Or would we add it to both RouteParentStatus and RouteParentStatusWithHostnames/HTTPRouteParentStatus?